resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/SslConfigurationProvider.scala

```diff
@@ -52,52 +60,66 @@ private[spark] class SslConfigurationProvider(
   private val sslSecretsDirectory = s"$DRIVER_CONTAINER_SECRETS_BASE_DIR/$kubernetesAppId-ssl"
 
   def getSslConfiguration(): SslConfiguration = {
-    val (driverSubmitSslOptions, isKeyStoreLocalFile) = parseDriverSubmitSslOptions()
-    if (driverSubmitSslOptions.enabled) {
-      val sslSecretsMap = mutable.HashMap[String, String]()
-      val sslEnvs = mutable.Buffer[EnvVar]()
-      val secrets = mutable.Buffer[Secret]()
-      driverSubmitSslOptions.keyStore.foreach(store => {
-        val resolvedKeyStoreFile = if (isKeyStoreLocalFile) {
-          if (!store.isFile) {
-            throw new SparkException(s"KeyStore specified at $store is not a file or" +
-              s" does not exist.")
-          }
-          val keyStoreBytes = Files.toByteArray(store)
-          val keyStoreBase64 = BaseEncoding.base64().encode(keyStoreBytes)
-          sslSecretsMap += (SUBMISSION_SSL_KEYSTORE_SECRET_NAME -> keyStoreBase64)
-          s"$sslSecretsDirectory/$SUBMISSION_SSL_KEYSTORE_SECRET_NAME"
-        } else {
-          store.getAbsolutePath
-        }
-        sslEnvs += new EnvVarBuilder()
-          .withName(ENV_SUBMISSION_KEYSTORE_FILE)
-          .withValue(resolvedKeyStoreFile)
-          .build()
-      })
-      driverSubmitSslOptions.keyStorePassword.foreach(password => {
+    val driverSubmitSslOptions = parseDriverSubmitSslOptions()
+    if (driverSubmitSslOptions.storeBasedSslOptions.enabled) {
+      val storeBasedSslOptions = driverSubmitSslOptions.storeBasedSslOptions
+      val keyStoreSecret = resolveFileToSecretMapping(
+          driverSubmitSslOptions.isKeyStoreLocalFile,
+          SUBMISSION_SSL_KEYSTORE_SECRET_NAME,
+          storeBasedSslOptions.keyStore,
+          "KeyStore")
```

Comment:
The casing is intentionally different I believe because in this case the type is mentioned at the beginning of the sentence of the error message and in the other case it's in the middle, hence requiring different grammar considerations in the choice of the string literal =P